hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieLogFileReader.java

```diff
@@ -160,12 +163,17 @@ private HoodieLogBlock readBlock() throws IOException {
       return createCorruptBlock(blockStartPos);
     }
 
-    // We may have had a crash which could have written this block partially
-    // Skip blockSize in the stream and we should either find a sync marker (start of the next
-    // block) or EOF. If we did not find either of it, then this block is a corrupted block.
-    boolean isCorrupted = isBlockCorrupted(blockSize);
-    if (isCorrupted) {
-      return createCorruptBlock(blockStartPos);
+    // Block is possibly corrupted only when write is not transactional
+    // the corrupted check could take 100's msec for larger file sizes, skipping the check for fs with transactional write
+    // see https://issues.apache.org/jira/browse/HUDI-2118
+    if (!StorageSchemes.isWriteTransactional(fs.getScheme())) {
```

Comment:
I'd suggest we fold this check into the `isBlockCorrupted` check itself:

- At the end of the day, this part of the same check
- We limit the scope of the change to just `isBlockCorrupted` method